dotCMS/src/main/java/com/dotmarketing/db/listeners/CommitAPI.java

```diff
@@ -0,0 +1,336 @@
+import com.google.common.annotations.VisibleForTesting;
+
+public class CommitAPI {
+
+
+    public enum CommitListenerStatus {
+        DISABLED, ALLOW_SYNC, ALLOW_ASYNC;
+    }
+    
+    private static class apiHolder {
+        static final CommitAPI api = new CommitAPI();
+    }
+
+    public static CommitAPI getInstance() {
+
+        return apiHolder.api;
+
+    }
+    /**
+     * Status for listeners of thread-local -based transactions. This allows to control as to which
+     * listeners are fired on a commit
+     */
+    private final ThreadLocal<CommitListenerStatus> listenersStatus = new ThreadLocal<CommitListenerStatus>() {
+        protected CommitListenerStatus initialValue() {
+            return CommitListenerStatus.ALLOW_ASYNC;
+        }
+    };
+
+
+    /**
+     * Returns the listeners status currently associated to the thread-local -based transaction
+     * (ALLOW_ASYNC by default)
+     */
+    public CommitListenerStatus status() {
+        return listenersStatus.get();
+    }
+
+    /**
+     * Allows to override the status of the listeners associated to the current thread-local -based
+     * transaction (DISABLED if overriden) When using TransactionListenerStatus.DISABLED, client code
+     * should be aware of controlling the operations that are suppossed to be done by listeners
+     * 
+     * @param status TransactionListenerStatus
+     */
+    public void setCommitListenerStatus(CommitListenerStatus status) {
+        listenersStatus.set(status);
+    }
+
+    /**
+     * Sets the commit listener status - convienience method
+     * 
+     * @param status
+     */
+    public void status(CommitListenerStatus status) {
+        this.setCommitListenerStatus(status);
+    }
+
+    final int COMMIT_LISTENER_QUEUE_SIZE;
+    final int COMMIT_LISTENER_THREADPOOL_SIZE;
+    final ExecutorService submitter;
+
+    private CommitAPI() {
+        COMMIT_LISTENER_QUEUE_SIZE = Config.getIntProperty("COMMIT_LISTENER_QUEUE_SIZE", 10000);
+        COMMIT_LISTENER_THREADPOOL_SIZE = Config.getIntProperty("COMMIT_LISTENER_THREADPOOL_SIZE", 10);
+        submitter = new ThreadPoolExecutor(1, COMMIT_LISTENER_THREADPOOL_SIZE, 30, TimeUnit.SECONDS,
+                        new LinkedBlockingDeque<Runnable>(COMMIT_LISTENER_QUEUE_SIZE));
+
+    }
+
+
+
+
+
+
+
+    @VisibleForTesting
+    static final ThreadLocal<Map<String, DotListener>> asyncCommitListeners = ThreadLocal.withInitial(LinkedHashMap::new);
+
+    @VisibleForTesting
+    static final ThreadLocal<Map<String, DotListener>> syncCommitListeners = ThreadLocal.withInitial(LinkedHashMap::new);
+
+    @VisibleForTesting
+    static final ThreadLocal<Map<String, DotListener>> rollbackListeners = ThreadLocal.withInitial(LinkedHashMap::new);
+
+
+    /**
+     * adds this to the end of async CommitListener list
+     */
+    public void addCommitListenerAsync(final String key, final Runnable runnable) {
+        if (!DbConnectionFactory.inTransaction())
+            return;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5639225)